src/aig-common/modules/standard/components/ipp-procedure/ipp-procedure-form/ipp-procedure-form.component.ts

```diff
@@ -0,0 +1,77 @@
+import { Component, OnInit } from '@angular/core';
+import { FormGroup, FormBuilder, Validators } from '@angular/forms';
+import { MatSnackBar } from '@angular/material/snack-bar';
+import { FuseProgressBarService } from '@fuse/components/progress-bar/progress-bar.service';
+import { ItalianPublicProcurementProcedureResourceService, ItalianPublicProcurementProcedureDTO  } from 'aig-standard';
+import { EventService } from 'aig-common/event-manager/event.service';
+
+@Component({
+    selector: 'aig-ipp-procedure-form',
+    templateUrl: './ipp-procedure-form.component.html',
+    styleUrls: ['./ipp-procedure-form.component.scss']
+})
+export class AigIppProcedureFormComponent implements OnInit {
+    constructor(
+        private _formBuilder: FormBuilder,
+        private _fuseProgressBarService: FuseProgressBarService,
+        private _snackBar: MatSnackBar,
+        private ippProcedureResourceService: ItalianPublicProcurementProcedureResourceService,
+        private eventService: EventService,
+    ) { }
+
+    private ippProcedureNewUpdateForm: FormGroup;
+    private step: any = {
+        form: true,
+        loading: false,
+        complete: false
+    };
+    
+    public ippProcedure: ItalianPublicProcurementProcedureDTO;
+
+    ngOnInit(): void {
+        this.ippProcedureNewUpdateForm = this._formBuilder.group({
+            id:[''],
+            name: ['', Validators.required],
+            code: ['', Validators.required],
+            wikiCode:['']
+        })
+    }
+
+    public createProcedure(){
+        if (!this.ippProcedureNewUpdateForm.valid) {
+            return;
+        }
+        this._fuseProgressBarService.show();
+        this.setStep("loading");
+
+        let ippProcedure: ItalianPublicProcurementProcedureDTO = {
```

Comment:
why do this..... simply:
`let ippProcedure: ItalianPublicProcurementProcedureDTO = this.ippProcedureNewUpdateForm.value;`